packages/docusaurus/bin/beforeCli.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const chalk = require('chalk');
+const fs = require('fs-extra');
+const semver = require('semver');
+const path = require('path');
+const updateNotifier = require('update-notifier');
+const boxen = require('boxen');
+
+const {
+  name,
+  version,
+  engines: {node: requiredVersion},
+} = require('../package.json');
+
+// Notify user if @docusaurus packages is outdated
+//
+// Note: this is a 2-step process to avoid delaying cli usage by awaiting a response:
+// - 1st run: trigger background job to check releases + store result
+// - 2nd run: display potential update to users
+//
+// Note: even if the
+//
+// cache data is stored in ~/.config/configstore/update-notifier-@docusaurus
+//
+const notifier = updateNotifier({
+  pkg: {
+    name,
+    version,
+  },
+  // Check is in background so it's fine to use a small value like 1h
+  // Use 0 for debugging
+  updateCheckInterval: 1000 * 60 * 60,
+  // updateCheckInterval: 0
+});
+
+// Hacky way to ensure we check for updates on first run
+// Note: the notification will only happen in the 2nd run
+// See https://github.com/yeoman/update-notifier/issues/209
+if (
+  !notifier.disabled &&
```

Comment:
@slorber I didn't find this field anywhere in the docs...

Reply:
this update notifier package is a bit complicated, you'd rather read source code to understand how it works 😅

Reply:
The reason is because I'm touching that source code a little and `ts-check` throws an error because this field is not in the typedef. That means it's not public API? I do see that exists, just not sure if it's actually useful...

Reply:
pretty sure we had failures without that check, I'd rather not change something that works just to please TS ;)

Reply:
Yeah, I clearly see that. I'm not saying to remove it, just curious how you dug it up if it's neither typedefed nor documented.